app/js/questionNav.js

```diff
@@ -1,55 +1,89 @@
     current++
-    let nextQuestion = document.querySelector(".q_" + current)
-    document.querySelector("h4").textContent = current + "/30"
-
-    if (current !== 1) {
-        document.querySelector(".prev").style.display = "block"
-    }
-    if (current === 30) {
-        document.querySelector(".next").style.display = "none"
-    }
-    if (nextQuestion !== null) {
-        document.querySelector(".active").classList.remove("active")
-        document.querySelector(".q_" + current).classList.add("active")
-    }
+    changeQuestion(current)
 }
 
 /**
-*adds active class to prev question, removes from current question
-* makes the prev and next buttons appear when needed
+ * adds active class to prev question, removes from current question
+ * makes the prev and next buttons appear when needed
  */
 function prev() {
     current--
-    let prevQuestion = document.querySelector(".q_" + current)
-    document.querySelector("h4").textContent = current + "/30"
+    changeQuestion(current)
+}
 
-    if (current === 1) {
-        document.querySelector(".prev").style.display = "none"
-        document.querySelector(".next").style.display = "block"
-    }
-    if (current !== 30) {
-        document.querySelector(".next").style.display = "block"
-    }
-    if (prevQuestion !== null) {
-        document.querySelector(".active").classList.remove("active")
-        document.querySelector(".q_" + current).classList.add("active")
+/**
+ * Takes you to any question based on parameter
+ *
+ * @param destinationPage is question to load
+ */
+function changeQuestion(destinationPage) {
+    let destinationQuestion = document.querySelector(".q_" + destinationPage)
+    let questionCount = document.querySelectorAll('#questions .question').length
+    let nextButton = document.querySelector(".next")
+    let prevButton = document.querySelector(".prev")
+    document.querySelector("h4").textContent = destinationPage + "/" + questionCount
+
+    switch(destinationPage) {
+        case 1:
+            prevButton.style.display = "none"
+            nextButton.style.display = "block"
+            break;
+        case questionCount:
+            prevButton.style.display = "block"
+            nextButton.style.display = "none"
+            break;
+        default:
+            prevButton.style.display = "block"
+            nextButton.style.display = "block"
     }
+
+    document.querySelector("#questions .active").classList.remove("active")
+    destinationQuestion.classList.add("active")
+    trackActiveQuestion(destinationPage)
+}
+
+/**
+ * Fills the question navbar with clickable elements that takes you to the given question number.
+ */
+function fillNav() {
+    let nav = document.querySelector("#question-nav")
+    let questions = document.querySelectorAll('.question')
+    let counter = 1
+    questions.forEach(function (question) {
+        nav.innerHTML += '<span class="nav-item unanswered-nav-box"><p>' + question.dataset['id'] + '</p></span>'
```

Comment:
p tags inside span tags is invalid html. if your gonna use an inline-block element use a div

Reply:
fixed in a420e9d